test/e2e-setup/bicep/modules/test-vm.bicep

```diff
@@ -0,0 +1,148 @@
+param adminUsername string = 'azureuser'
+
+@description('SSH public key for the VM')
+param sshPublicKey string
+
+@description('VM size')
+param vmSize string = 'Standard_B2s'
+
+@description('Location for the VM')
+param location string = resourceGroup().location
+
+//
+// EXISTING RESOURCES
+//
+
+resource vnet 'Microsoft.Network/virtualNetworks@2022-07-01' existing = {
+  name: vnetName
+}
+
+resource subnet 'Microsoft.Network/virtualNetworks/subnets@2022-07-01' existing = {
+  name: subnetName
+  parent: vnet
+}
+
+//
+// NETWORK RESOURCES
+//
+
+resource publicIP 'Microsoft.Network/publicIPAddresses@2022-07-01' = {
+  name: '${vmName}-pip'
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    publicIPAllocationMethod: 'Static'
+    publicIPAddressVersion: 'IPv4'
+  }
+}
+
+resource nic 'Microsoft.Network/networkInterfaces@2022-07-01' = {
+  name: '${vmName}-nic'
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig1'
+        properties: {
+          subnet: {
+            id: subnet.id
+          }
+          privateIPAllocationMethod: 'Dynamic'
+          publicIPAddress: {
+            id: publicIP.id
+          }
+        }
+      }
+    ]
+  }
+}
+
+//
+// VIRTUAL MACHINE
+//
+
+resource vm 'Microsoft.Compute/virtualMachines@2023-03-01' = {
+  name: vmName
+  location: location
+  properties: {
+    hardwareProfile: {
+      vmSize: vmSize
+    }
+    osProfile: {
+      computerName: vmName
+      adminUsername: adminUsername
+      linuxConfiguration: {
+        disablePasswordAuthentication: true
+        ssh: {
+          publicKeys: [
+            {
+              path: '/home/${adminUsername}/.ssh/authorized_keys'
+              keyData: sshPublicKey
+            }
+          ]
+        }
+      }
+    }
+    storageProfile: {
+      imageReference: {
+        publisher: 'Canonical'
+        offer: '0001-com-ubuntu-server-jammy'
+        sku: '22_04-lts-gen2'
+        version: 'latest'
+      }
+      osDisk: {
+        createOption: 'FromImage'
+        managedDisk: {
+          storageAccountType: 'Premium_LRS'
+        }
+      }
+    }
+    networkProfile: {
+      networkInterfaces: [
+        {
+          id: nic.id
+        }
+      ]
+    }
+  }
+}
+
+//
+// VM EXTENSION FOR TOOLS
+//
+
+resource vmExtension 'Microsoft.Compute/virtualMachines/extensions@2023-03-01' = {
+  name: 'installTools'
+  parent: vm
+  location: location
+  properties: {
+    publisher: 'Microsoft.Azure.Extensions'
+    type: 'CustomScript'
+    typeHandlerVersion: '2.1'
+    autoUpgradeMinorVersion: true
+    settings: {}
+    protectedSettings: {
+      commandToExecute: 'sh -c "export DEBIAN_FRONTEND=noninteractive && rm -rf /var/lib/apt/lists/* && apt-get update && apt-get install -y curl wget && curl -LO https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl && install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl"'
```

Comment:
Could we use RHEL image with openshift build of kubectl to avoid relying on binary artefacts from dl.k8s.io? If there is no image let's just prepare one and host it on quay and/or in our acr. This will reduce risks of flakynes caused by issues of external infra.